### PR TITLE
Add extra meta to about.json

### PR DESCRIPTION
**Pull request by cjmartian**
Allows arbitrary information to be added to about.json
This might be helpful for embedding build system log IDs into the conda package for better traceability.

### CODE REVIEW

File: conda_build/build.py [hunk 1]
```diff
@@ -1179,10 +1179,17 @@ def write_about_json(m):
             d['conda_private'] = conda_private
         except (KeyError, AttributeError):
             pass
+        # Don't want to assume git is installed
+        # or that we are even in a git repo
+        # Adding this to extra since its arbitrary info
+        extra = m.get_section('extra')
+        # Add burn-in information to extra
+        if m.config.extra_meta:
+            extra.update({k: v for k, v in m.config.extra_meta.items()})
```
Comment (jezdez) [inline conda_build/build.py:1188]:
```suggestion
            extra.update(m.config.extra_meta)
```

File: conda_build/build.py [hunk 2]
```diff
@@ -1179,10 +1179,17 @@ def write_about_json(m):
             d['conda_private'] = conda_private
         except (KeyError, AttributeError):
             pass
+        # Don't want to assume git is installed
+        # or that we are even in a git repo
```
Comment (jezdez) [inline conda_build/build.py:1183]:
```suggestion
```

File: conda_build/cli/main_build.py [hunk 3]
```diff
@@ -354,7 +355,13 @@ def parse_args(args):
                    help=('Extra dependencies to add to all environment creation steps.  This '
                          'is only enabled for testing with the -t or --test flag.  Change '
                          'meta.yaml or use templates otherwise.'), )
-
+    p.add_argument(
+        '--extra-meta',
+        nargs='*',
+        action=KeyValueAction,
+        help="Key value pairs of metadata to add to about.json. Should be "
+        "defined as Key=Value with a space separating each pair.",
```
Comment (jezdez) [inline conda_build/cli/main_build.py:363]:
```suggestion
        "defined as Key=Value with a space separating each pair.",
        metavar="KEY=VALUE",
```

**Review by jezdez — APPROVED** [review]

**Comment by cjmartian** [comment]
@chenghlee added a test for this and https://github.com/conda/conda-build/pull/4304

**Review by dbast — APPROVED** [review]

**Pull request merged**